node/src/effect.rs

```diff
@@ -196,6 +192,48 @@ pub(crate) struct Responder<T> {
     is_shutting_down: SharedFlag,
 }
 
+/// A responder that will automatically send a `None` on drop.
+#[must_use]
+#[derive(DataSize, Debug)]
+pub(crate) struct AutoClosingResponder<T>(Responder<Option<T>>);
```

Comment:
How about using an `enum`? 
```rust
enum ResponderResult<T> {
  Dropped,
  Ok(T),
}
```
? 

The `None` case of the `Option` has some well-defined meaning - absence of an item. Current encoding also makes it bit weird if `T = Option<M>`, we would get `Some(None)` or `Some(Some(_))`.

Reply:
I tend to agree, but turns out we're using the `None` variant in a couple of places already where a function returns an error, and we don't have valid data to send via `respond()`.  So it would be inappropriate to replace `None` at such callsites with `ResponderResult::Dropped`.

I don't really like that, but I tried changing it to work "properly" by swapping the `Option<T>` for a `Result<T, AutoClosingResponderError<E>>`, and this affects way too much code to be included in this PR.

For now, I think it's best to leave this as is.